reference-implementation/lib/abstract-ops/transform-streams.js

```diff
@@ -221,32 +231,61 @@ function TransformStreamDefaultSinkWriteAlgorithm(stream, chunk) {
 }
 
 function TransformStreamDefaultSinkAbortAlgorithm(stream, reason) {
-  // abort() is not called synchronously, so it is possible for abort() to be called when the stream is already
-  // errored.
-  TransformStreamError(stream, reason);
-  return promiseResolvedWith(undefined);
+  verbose('TransformStreamDefaultSinkAbortAlgorithm()');
+
+  const controller = stream._controller;
+  if (controller._finishPromise) {
```

Comment:
```suggestion
  if (controller._finishPromise !== undefined) {
```
We try to be very explicit in the reference implementation. 😉